CRUD endpoints for Courses

s7u4rt99 opened the pull request:
Resolves #36

BryannYeap commented on pages/api/courses/index.ts:
```diff
@@ -0,0 +1,30 @@
+        }
+    } catch (error) {
+        console.log(error)
+        res.status(400).json({ message: error })
```
I believe we should default to a status code of 500 here, as any unhandled errors here is the server's fault, since the client has already sent a valid request. If the client sends a request with an invalid JSON body, it already seems to automatically give a 400 status code.

s7u4rt99 replied:
done

BryannYeap commented on pages/api/courses/[id].ts:
```diff
@@ -0,0 +1,44 @@
+        }
+    } catch (error) {
+        console.log(error)
+        res.status(400).json({ message: error })
```
Same comment as in `index.ts`

s7u4rt99 replied:
done

BryannYeap reviewed (commented):
Almost LGTM! Just a few nits.

s7u4rt99 commented on pages/api/courses/[id].ts:
```diff
@@ -0,0 +1,44 @@
+        const id = req.query.id as string
+
+        if (httpMethod == 'GET') {
+            const courses = await prisma.course.findFirst({
```
done

BryannYeap reviewed (approved):
LGTM!

kevinchua6 reviewed (changes requested):
Looks good, just follow the message and data response conventions given in https://github.com/Project-Juniors-Club/jc-app/pull/53, and update postman also

weixin-tan commented:
Take note of #60, might be able to do within this PR. Or separate if the migration is not too tricky.

s7u4rt99 commented:
Added message and data responses, and incorporated with the extra fields added to courses on main. Not sure what is needed to add to postman @kevinchua6

kevinchua6 commented:
Looks good, can you update postman with a sample body with name, description etc so I can test it, thanks!
![image](https://user-images.githubusercontent.com/45702380/198203614-d1631d3c-9e96-4a1a-9e07-7260e0c3d9b5.png)

s7u4rt99 commented:
done @kevinchua6

kevinchua6 reviewed (approved):
LGTM
